blog/2023-02-05-github-slack/index.md

```diff
@@ -0,0 +1,221 @@
+
+​    This article will show how to do this in 5 minutes on Playground with Vanus and Vanus Connect， the results are shown below:
+
+![GitHub-to-Slack-Result](./img/GitHub-to-Slack-Result.gif)
+
+## Prerequisites：
+
+- Playground： An online k8s environment where Vanus can be deployed.
+- GitHub： Your open-source repository
+- slack： A working slack account
+
+## Part 1: Deploying Vanus in the playground
+
+1 Enter the login page and click the Continue with Github button to log in with the GitHub account
+
+![playground-login](./img/playground-login.jpeg)
+
+2  Wait for the automatic deployment of Kubernetes to complete, about the 30s.
+
+![playground](./img/playground.jpeg)
+
+3 Deploy Vanus to the terminal on the right side of the web page
+
+```Shell
+kubectl apply -f https://vanus.s3.us-west-2.amazonaws.com/releases/v0.4.0/vanus.yaml
+```
+
+ Verify: watch -n2 kubectl get po -n Vanus，
+
+```Plain
+ $ watch -n2 kubectl get po -n vanus
+vanus-controller-0               1/1     Running   0          96s
+vanus-controller-1               1/1     Running   0          72s
+vanus-controller-2               1/1     Running   0          69s
+vanus-gateway-8677fc868f-rmjt9   1/1     Running   0          97s
+vanus-store-0                    1/1     Running   0          96s
+vanus-store-1                    1/1     Running   0          68s
+vanus-store-2                    1/1     Running   0          68s
+vanus-timer-5cd59c5bf-hmprp      1/1     Running   0          97s
+vanus-timer-5cd59c5bf-pqkd5      1/1     Running   0          97s
+vanus-trigger-7685d6cc69-8jgsl   1/1     Running   0          97s
+```
+
+4 Install vsctl (the command line tool)
+
+```Plain
+curl -O https://vsctl.s3.us-west-2.amazonaws.com/releases/v0.4.0/linux-amd64/vsctl
+chmod ug+x vsctl
+mv vsctl /usr/local/bin
+```
+
+5  Set the endpoint for vsctl to access vanus
+
+```Plain
+export VANUS_GATEWAY=192.168.49.2:30001
+```
+
+6 Create eventbus
+
+```Plain
+$ vsctl eventbus create  GitHub-slack
++----------------+-------------+
+|     RESULT     |   EVENTBUS  |
++----------------+-------------+
+| Create Success | github-slack|
++----------------+-------------+
+```
+
+## Part 2: Deploy the GitHub source connector
+
+7 Set config file
+
+ Create config.yml in any directory, the content is as follows
+
+```Plain
+{
+  "v_target": "http://192.168.49.2:30001/gateway/github-slack",
+  "v_port": "8082"
+}
+```
+
+8 Deploy the GitHub source connector and run the following command in the same directory:
+
+```Plain
+docker run --network=host -v $(pwd)/config.json:/vance/config/config.json  --rm vancehub/source-github > a.log &
+```
+
+## Part 3: Creating a Slack app
+
+10 Create a slack app
+
+First, log in to slack, and click Create New APP,then select From Scrath, And fill in the App Name, select the corresponding Workspace
+
+![create-slack-app](./img/create-slack-app.jpeg)
+
+11 Select Incoming webhook，and click on
```

Comment:
Creating an incoming webhook isnt needed for the slack sink. We should remove it.